toolchain/check/function.cpp

```diff
@@ -191,60 +191,88 @@ auto CheckFunctionTypeMatches(Context& context,
                      context.functions().Get(prev_function_id), substitutions);
 }
 
+// Checks to see if a structurally valid redeclaration is allowed in context.
+// These all still merge.
+static auto CheckIsAllowedRedecl(Context& context, Parse::NodeId node_id,
+                                 SemIR::Function& new_function,
```

Comment:
I expect we'll eventually want to factor this out and use the same implementation for classes, impls, and interfaces. No need to do that now, but I suppose we should start thinking about how to provide a way to access the common fields of `SemIR::Function`, `SemIR::Class`, etc.